src/server.cpp

```diff
@@ -20,19 +23,55 @@ void Server::start_receive()
         std::bind(&Server::handle_receive, this, pls::_1, pls::_2));
 }
 
-void Server::handle_receive(const error_code& error, std::size_t /*bytes_transferred*/)
+void Server::handle_receive(const error_code& error, std::size_t bytes)
 {
-    if (!error || error == boost::asio::error::message_size)
+    if (!error)
     {
-        const std::string message("Hello, World!");
+        size_t length;
+        try
+        {
+            length = recv_buffer_.at(2) * 256 + recv_buffer_.at(3);
+        }
+        catch (std::out_of_range const& exc)
+        {
+            std::cout << exc.what() << "\n";
+        }
 
-        socket_.async_send_to(boost::asio::buffer(message), remote_endpoint_,
-          std::bind(&Server::handle_send, this, message,pls::_1, pls::_2));
+        if (bytes >= length)
+        {
+            uint8_t send_code;
 
-        start_receive();
+            if (recv_buffer_[0] == ACCESS_REQUEST)
+                send_code = ACCESS_ACCEPT;
+
+            send_buffer_[0] = send_code;
```

Comment:
If `recv_buffer_[0]` is not `ACCESS_REQUEST`, `send_code` would be undefined. If you don't want to handle anything other than `ACCESS_REQUEST` - just do some logging and do nothing.

Reply:
Also, what is the purpose of the `send_code` variable? Why can't you just assign `send_buffer_[0]` the value directly?